plugins/lookup/secretsmanager_random_password.py

```diff
@@ -0,0 +1,182 @@
+"""
+
+EXAMPLES = r"""
+  - name: generate random password
+    debug: msg="{{ lookup('secretsmanager_random_password') }}"
+
+  - name: generate random 12-character password without punctuation
+    debug: msg="{{ lookup('secretsmanager_random_password', 12, exclude_punctuation=True) }}"
+
+  - name: create a secret using a random password
+    community.aws.secretsmanager_secret:
+      name: 'test_secret_string'
+      state: present
+      secret_type: 'string'
+      secret: "{{ lookup('secretsmanager_random_password') }}"
+"""
+
+RETURN = r"""
+_raw:
+  description:
+    Returns the random password.  This password is not saved and will always be new.
+"""
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AWSLookupBase
+
+from ansible.errors import AnsibleLookupError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import string_types
+from ansible.module_utils.six import integer_types
+
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
+from ansible_collections.amazon.aws.plugins.plugin_utils.lookup import AWSLookupBase
+
+
+class LookupModule(AWSLookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        """
+        :param terms: a list containing the password length
+            e.g. ['example_secret_name', 'example_secret_too' ]
+        :param variables: ansible variables active at the time of the lookup
+        :returns: A list of parameter values or a list of dictionaries if bypath=True.
+        """
+
+        super().run(terms, variables, **kwargs)
+
+        # validate argument terms
+        if len(terms) > 1:
+            raise AnsibleLookupError("secretsmanager_random_password must have zero or one argument")
+
+        on_denied = self.get_option("on_denied")
+
+        # validate arguments 'on_denied'
+        if on_denied is not None and (
+            not isinstance(on_denied, string_types) or on_denied.lower() not in ["error", "warn", "skip"]
+        ):
+            raise AnsibleLookupError(
+                f'"on_denied" must be a string and one of "error", "warn" or "skip", not {on_denied}'
+            )
+
+        params = {}
+        # validate password length argument or option
+        self.debug(f"Options: {self.get_options()}")
+        password_length = self.get_option("password_length")
+        if len(terms) == 1:
+            if password_length is not None:
+                raise AnsibleLookupError('"password_length" should be provided as argument or keyword, not both')
+            password_length = terms[0]
+        if password_length is not None:
+            if not isinstance(password_length, integer_types) or password_length < 1:
+                raise AnsibleLookupError('"password_length" must be an integer greater than zero, if provided')
+            params["PasswordLength"] = password_length
```

Comment:
I'm not a big fan of this logic.  I understand why you want to be able to do  `lookup('community.aws.secretsmanager_random_password', 12)`, but combining this with a conflicting parameter which does something identical can be confusing and is just additional logic which needs testing and can break.

I'd recommend dropping support for pulling this from terms.  You're not pulling a secret called "12" from the lookup, you're just passing it a length.

Reply:
I've updated it per your recommendation.